private/rules/java_export.bzl

```diff
@@ -235,13 +239,16 @@ def maven_export(
         testonly = testonly,
     )
 
+    classifier_artifacts = dict(classifier_artifacts)  # unfreeze
+    classifier_artifacts.setdefault("sources", ":%s-maven-source" % name)
+    classifier_artifacts.setdefault("javadoc", docs_jar)
```

Comment:
Neither `sources` not `javadoc` are mandatory in some cases (eg. when publishing a BOM), so please check that these values are set before adding them to the `dict`.